src/ontodia/editor/editorController.tsx

```diff
@@ -651,29 +662,25 @@ export class EditorController {
         const model = elements[0].data;
 
         const event = state.index.elements.get(elementIri);
-        if (event && event.type === AuthoringKind.ChangeElement) {
-            if (event.before) {
-                // restore initial data
-                this.model.history.execute(
-                    setElementData(this.model, elementIri, event.before)
-                );
-            } else {
-                // delete newly created entity
-                for (const element of elements) {
-                    this.model.removeElement(element.id);
+        const isOriginalEntity = !event || event.type === AuthoringKind.ChangeElement && event.before;
```

Comment:
I think this block could be simplified by reusing `discardChange()` logic a little bit more:
```
// remove new connected links
const linksToRemove = new Set<string>();
for (const element of elements) {
    for (const link of element.links) {
        if (link.data && AuthoringState.isNewLink(state, link.data)) {
            linksToRemove.add(link.id);
        }
    }
}
// delete separately to prevent modification while iterating
linksToRemove.forEach(linkId => this.model.removeLink(linkId));

if (event) {
    this.discardChange(event);
}
```